src/main/scala/net/ripe/rpki/publicationserver/Boot.scala

```diff
@@ -99,11 +109,12 @@ class PublicationServerApp(conf: AppConfig, https: ConnectionContext, logger: Lo
   }
 
   def shutdown() = {
-       Await.result(httpBinding, 10.seconds)            
-            .terminate(hardDeadline = 3.seconds)
-            .flatMap(_ => 
-                Await.result(httpsBinding, 10.seconds)            
-                .terminate(hardDeadline = 3.seconds))
-            .flatMap(_ => system.terminate())
+      this.repositoryWriter.cancel()
+      Await.result(httpBinding, 10.seconds)
+           .terminate(hardDeadline = 3.seconds)
+           .flatMap(_ =>
+                Await.result(httpsBinding, 10.seconds)
+                     .terminate(hardDeadline = 3.seconds))
+           .flatMap(_ => system.terminate())
```

Comment:
Note that the change here is `this.repositoryWriter.cancel()`. The changes on `Await.result ..` lines are correction of too much indentation.